Source/WebCore/bindings/js/JSDOMGlobalObject.cpp

```diff
@@ -390,6 +391,13 @@ bool JSDOMGlobalObject::canCompileStrings(JSGlobalObject* globalObject, Compilat
     return result.releaseReturnValue();
 }
 
+Structure* JSDOMGlobalObject::trustedScriptStructure(JSGlobalObject* globalObject)
+{
+    auto& thisObject = static_cast<JSDOMGlobalObject&>(*globalObject);
+
+    return getDOMStructure<JSTrustedScript>(globalObject->vm(), thisObject);
```

Comment:
It seems that the dom wrapper does not keep this alive

Reply:
We should either add a WriteBarrier<> and mark it in visitChildrenImpl, or make it a weakref